src/components/MarkerPopup/MarkerPopup.jsx

```diff
@@ -1,61 +1,80 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
-import { Marker, Popup, useMap } from 'react-leaflet';
+import { Marker, Popup } from 'react-leaflet';
 import { isMobile } from 'react-device-detect';
+import { httpService } from '../../services/http/httpService';
+import styled from 'styled-components';
 
-import { MarkerContent } from './MarkerContent';
+import { LocationDetailsBox } from './LocationDetails';
 import { MobilePopup } from './MobilePopup';
 
-const MobileMarker = ({ place }) => {
-    const [open, setOpen] = useState(false);
-    const map = useMap();
+const StyledPopup = styled(Popup)`
+    min-width: 300px;
+`;
 
-    const handleClickOpen = () => {
-        const offset = 0.003;
-        const newLat = place.position[0] - offset;
-        map.panTo([newLat, place.position[1]], { duration: 0.5 });
-        setOpen(true);
-    };
+export const AutoOpenPopup = ({ children }) => {
+    const popupRef = useRef(null);
 
-    const handleClose = () => {
-        setOpen(false);
-    };
+    useEffect(() => {
+        if (popupRef.current) {
+            const marker = popupRef.current._source;
+            if (marker) {
+                marker.openPopup();
+            }
+        }
+    }, []);
 
-    return (
-        <Marker
-            position={place.position}
-            key={place.metadata.UUID}
-            eventHandlers={{ click: handleClickOpen }}
-        >
-            <MobilePopup isOpen={open} onCloseHandler={handleClose}>
-                <MarkerContent place={place} isMobileVariable={true} />
-            </MobilePopup>
-        </Marker>
-    );
+    return <StyledPopup ref={popupRef}>{children}</StyledPopup>;
 };
 
-const DesktopMarker = ({ place }) => {
+const DesktopPopup = AutoOpenPopup;
+
+const LocationDetailsBoxWrapper = ({ theplace }) => {
+    const [place, setPlace] = useState(null);
+    const ChosenPopup = isMobile ? MobilePopup : DesktopPopup;
+
+    if (!window.USE_LAZY_LOADING) {
+        return (
+            <ChosenPopup>
+                <LocationDetailsBox place={theplace} />
+            </ChosenPopup>
+        );
+    }
+
+    useEffect(() => {
+        const fetchPlace = async () => {
+            const fetchedPlace = await httpService.getLocation(theplace.UUID);
+            setPlace(fetchedPlace);
+        };
+        fetchPlace();
+    }, [theplace.UUID]);
+
     return (
-        <Marker position={place.position} key={place.metadata.UUID}>
-            <Popup>
-                <MarkerContent place={place} isMobileVariable={false} />
-            </Popup>
-        </Marker>
+        <ChosenPopup>
+            {place ? <LocationDetailsBox place={place} /> : <p>Loading...</p>}
+        </ChosenPopup>
     );
```

Comment:
That pattern is called `early exit`  to not have so many braces and indentations.  (To get shortest exit from function on beginning)

Reply:
Please separate comment which serve different things, but to answer your question: yes, probably. But is there any reason to do so?

Reply:
It would make sense to me, because fetching point data is something that happens when you do action (clicking) upon Marker. `LocationDetailsBoxWrapper` is some abstraction

Reply:
To be honest... I also think so, but that is not what author of `leaflet-react` thought. IMO we could get rid of `Popup` object and create our own Popup (like you did with <Dialog>.

But if so - we should do it in another issue.

Reply:
Agreed
https://github.com/Problematy/goodmap-frontend/issues/74